pkg/database/gorm.go

```diff
@@ -1,31 +1,48 @@
 package database
 
 import (
+	"context"
+	"fmt"
+
 	"github.com/DATA-DOG/go-txdb"
-	"github.com/jinzhu/gorm"
+	"gorm.io/driver/mysql"
+	"gorm.io/gorm"
+)
 
-	// Imports required gorm MySQL dialect.
-	_ "github.com/jinzhu/gorm/dialects/mysql"
+const (
+	testDBDriver = "txdb"
+	testEnv      = "test"
 )
 
 // NewConnection returns a new Gorm database connection.
 func NewConnection(config *Config, environment string) (*gorm.DB, error) {
-	var db *gorm.DB
-	var err error
-
 	connectionDetails := NewConnectionDetails(config)
 
-	switch environment {
-	case "test":
-		txdb.Register("txdb", connectionDetails.Dialect, connectionDetails.String())
-		db, err = gorm.Open(connectionDetails.Dialect, "txdb", "tx_1")
-	default:
-		db, err = gorm.Open(connectionDetails.Dialect, connectionDetails.String())
+	connectionString := connectionDetails.String()
+	driverName := connectionDetails.Dialect
+
+	if environment == testEnv {
+		txdb.Register(testDBDriver, connectionDetails.Dialect, connectionString)
+		driverName = testDBDriver
 	}
 
-	if err == nil {
-		db.DB().SetMaxIdleConns(connectionDetails.Pool)
+	dialector := mysql.New(mysql.Config{
+		DSN:        connectionString,
+		DriverName: driverName,
+	})
+
+	db, err := gorm.Open(dialector, nil)
+	if err != nil {
+		return nil, fmt.Errorf("opening gorm connection. ERR: %w", err)
```

Comment:
It's always good to wrap the error. This way, we can keep track of the error chain and it helps with readability. Also finding the problem by reading the log makes it easier.

Let's say you have 2 functions and in both, you are reading a file, if you don't wrap them, in case of error, let's imagine "file not found", you'll get same error on both files; but if you wrap them, you will have something like:
- "func1 couldn't fetch file, err: io.readfile, err: file not found"
- "func2 couldn't fetch file, err: io.readfile, err: file not found"
instead of:
- "err: file not found"
- "err: file not found"

Reply:
> Usually, errors are wrapped on the consumer side

Is it a best practice I don't know? Is a code dependent to it being consumer or producer?

> we don't need additional context here. 

you agree that "wrapping up the errors for more context can be helpful" and then you say we don't need it?

I cannot connect the dots here. I wrote down an example case with full context why it's useful, now if you agree it's helpful, how come it seems redundant :) .

Reply:
I digged deeper and must say you are right, wrapping errors from api interface is not a common practice. I will revert my changes.
thanks for the learning :)